cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java

```diff
@@ -623,6 +627,50 @@ private boolean processAddNodeLocally(Node node, AddNodeResponse response,
         return true;
       }
 
+      // check status of the new node
+//      AsyncClient client = (AsyncClient) connectNode(node);
+//      try {
+//        AsyncMethodCallback<String> result = new AsyncMethodCallback<String>() {
+//          String words;
+//          @Override
+//          public void onComplete(String s) {
+//            this.words = s;
+//          }
+//          @Override
+//          public void onError(Exception e) {
+//            System.out.println("some error happens");
+//          }
+//          public String getWords() {
+//            return words;
+//          }
+//        };
+//        System.out.println(result);
+//        client.echo("hello world", result);
+//      } catch (TException e) {
+//        e.printStackTrace();
+//      }
+      CheckStatusRequest checkStatusRequest = new CheckStatusRequest();
+      checkStatusRequest.setHashSalt(ClusterConstant.HASH_SALT);
+      checkStatusRequest
+          .setPartitionInterval(IoTDBDescriptor.getInstance().getConfig().getPartitionInterval());
+      checkStatusRequest.setReplicationNumber(config.getReplicationNum());
+      CheckStatusHandler checkStatusHandler = new CheckStatusHandler();
+      try {
+        sendStatusToNewNode(node, checkStatusRequest, checkStatusHandler);
+      } catch (TException exception) {
+        logger.error("Failed to send current state to the new node {}", node, exception);
+      }
+      if (!checkStatusHandler.getCheckStatusResponse().isPartitionalIntervalEquals()) {
+        logger.info("The partition interval of the new node {} conflicts.", node);
+        return true;
+      } else if (!checkStatusHandler.getCheckStatusResponse().isHashSaltIntervalEquals()) {
+        logger.info("The hash salt of the new node {} conflicts.", node);
+        return true;
+      } else if (!checkStatusHandler.getCheckStatusResponse().isReplicationNumEquals()) {
+        logger.info("The replication number of the new node {} conflicts.", node);
+        return true;
+      }
```

Comment:
I think it is simpler to just add these configurations that should be checked into the `AddNodeRequest` so that you can cut one communication down.
Besides, please remove the commented code.

Reply:
That's a good idea. I will change it